### PR TITLE
Onwards and upwards

### PR DESCRIPTION
**Pull request by armanbilge**
Bumps Scala 3 to 3.1.x and Scalacheck to 1.16.x. Closes https://github.com/typelevel/scalacheck/issues/878. Supersedes https://github.com/typelevel/scalacheck/pull/868.

### CODE REVIEW

**Review by rossabaker — APPROVED** [review]

**Comment by adampauls** [comment]
I'm dying of curiosity: what is stopping this PR from being merged? Lots of upstream stuff seems to be blocked on it?

**Comment by armanbilge** [comment]
@adampauls see the extensive discussion in https://github.com/typelevel/cats/pull/4016. So, historically it's been a controversial issue, although it may not be as much anymore.

FWIW my mind has been stuck on this PR too :) any @typelevel/scalacheck maintainer feel inclined to merge, or at least add another approval?

**Comment by SethTisue** [comment]
I've asked on #878 for some input from the Center and/or Virtus

**Comment by julienrf** [comment]
Please note that Scala 3.1.2 will support `-scala-output-version`. Would you consider using this feature to emit Scala 3.0-compatible TASTy files?

**Pull request merged**
